pytorch_lightning/utilities/fetching.py

```diff
@@ -172,8 +175,6 @@ def _attach_data_fetcher_fn(loader: DataLoader) -> None:
 
     def __iter__(self) -> "AbstractDataFetcher":
         self.reset()
-        self._attach_data_fetcher()
-        _patch_dataloader_get_iterators()
```

Comment:
I believe these only need to be called once on setup, so moved them

Reply:
If the iterators are being re-created, we won't be patching the correct new ones.

Reply:
But you might be right. Worth to check.

Reply:
`_patch_dataloader_get_iterators` is a simple monkeypatch on the class unrelated to the instances

and `_attach_data_fetcher` only uses the `self.dataloader` property which is read-only and only gets set during `setup` (the place where we call these 2 now)